cumulusci/tasks/bulkdata.py

```diff
@@ -797,6 +799,313 @@ def _sqlite_dump(self):
                 f.write(line + "\n")
 
 
+def represent_ordereddict(dumper, data):
+    value = []
+
+    for item_key, item_value in data.items():
+        node_key = dumper.represent_data(item_key)
+        node_value = dumper.represent_data(item_value)
+
+        value.append((node_key, node_value))
+
+    return yaml.nodes.MappingNode(u"tag:yaml.org,2002:map", value)
+
+
+yaml.SafeDumper.add_representer(OrderedDict, represent_ordereddict)
+
+
+class GenerateMapping(BaseSalesforceApiTask):
+    task_docs = """
+    Generate a mapping file for use with the `extract_dataset` and `load_dataset` tasks.
+    This task will examine the schema in the specified org and attempt to infer a
+    mapping suitable for extracting data in packaged and custom objects as well as
+    customized standard objects.
+
+    Mappings cannot include reference cycles - situations where Object A refers to B,
+    and B also refers to A. Mapping generation will fail for such data models; to
+    resolve the issue, specify the `ignore` option with the name of one of the
+    involved lookup fields to suppress it. `ignore` can be specified as a list in
+    `cumulusci.yml` or as a semicolon-separated string at the command line.
+
+    In most cases, the mapping generated will need minor tweaking by the user. Note
+    that the mapping omits features that are not currently well supported by the
+    `extract_dataset` and `load_dataset` tasks, including self-lookups and references to
+    the `User` object.
+    """
```

Comment:
Don't forget to run `cci task doc > docs/tasks.rst` to update the task documentation.

Reply:
(We'll also want some updates to the narrative documentation in https://cumulusci.readthedocs.io/en/latest/bulk_data.html -- that can happen in another pull request.)